lib/forms.js

```diff
@@ -13,8 +15,63 @@ var dataTargetImpl = require('./impl/dataTargets');
 var config = require('./config');
 var logger = logfns.getLogger();
 
+function getDbName(uri) {
+  var parsedMongooseUri = mongoUriParser.parse(uri);
+  var dbName = parsedMongooseUri.database;
+  return dbName;
+}
+
+/**
+ * Get a new mongoose connection instance. Will reuse an existing mongoose connection if it's passed in (this connection should be authenticated already and have access to the target db. E.g. admin user).
+ * Otherwise a new mongoose connection (connection pool) will be created.
+ * @param {string} url the uri of the mongodb
+ * @param {object} mongooseConnection an existing mongoose connection. If exists, it will be resued for the mongoose connection
+ * @param {function} cb the callback function
+ */
+function getMongooseConnection(uri, mongooseConnection, cb) {
```

Comment:
Might want to move this function into a different file. This file is very overloaded as it is.